Capstone/Friends/FriendDetailsViewController.m

```diff
@@ -0,0 +1,128 @@
+//
+//  FriendDetailsViewController.m
+//  Capstone
+//
+//  Created by Abel Kelbessa on 8/1/22.
+//
+#import <Parse/Parse.h>
+#import "ParseCollege.h"
+#import "Parse/PFImageView.h"
+#import "FriendDetailsViewController.h"
+#import "FriendMatchesCell.h"
+#import "DetailsViewController.h"
+ 
+@interface FriendDetailsViewController ()<UITableViewDelegate, UITableViewDataSource>
+@property (weak, nonatomic) IBOutlet UILabel *username;
+@property (weak, nonatomic) IBOutlet UIButton *sendFriendRequest;
+@property (weak, nonatomic) IBOutlet UIButton *cancelFriendRequest;
+@property (weak, nonatomic) IBOutlet UIImageView *userImage;
+@property (weak, nonatomic) IBOutlet UIButton *friends;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@end
+ 
+@implementation FriendDetailsViewController {
+    NSArray *usersMatchedColleges;
+    NSMutableArray *friends;
+    NSMutableArray *localFriends;
+    PFRelation *matchesRelation;
+    PFUser *currentUser;
+}
+ 
+- (void)viewDidLoad {
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    self.username.text = self.user.username;
+    self->localFriends = [[NSMutableArray alloc] init];
+    self->currentUser = [PFUser currentUser];
+    self->localFriends = currentUser[@"friends"];
+    self.friends.hidden = YES;
+    self.tableView.hidden = YES;
+    [self requestStatusChecker];
+    [self getUserImage];
+    [self getUsersMatches];
+}
+ 
+- (void)getUserImage {
+    PFFileObject *userProfileImage = self.user[@"image"];
+    [userProfileImage getDataInBackgroundWithBlock:^(NSData *imageData, NSError *error) {
+        if (!error) {
```

Comment:
Since this is a UI update, we should put this in the main queue: https://riptutorial.com/ios/example/27493/getting-the-main-queue

The code works here because `getUserImage` is called on `viewDidLoad` (which occurs before `viewDidAppear`), so we don't have any race conditions that would cause the image to not be loaded; however, things can get accidentally complicated if you ever needed to call `getUserImage` again. For now, let's just follow good practice!

Reply:
makes sense